### PR TITLE
Support custom analyzer in index setting

### PR DESCRIPTION
**Pull request by CCheSumo**
[x] Added support for custom index setting 
including
- [x] number of shards, replicas 
- [x] provide custom analyzer 

[x] Updated autocomplete interface to accept custom analyzer
- [x] default to previous behavior for backward compatibility

[x] Added unit tests for 
- [x] set up index setting
- [x] set up document mapping
- [x] case insensitive autocomplete with custom analyzer with special characters
- [x] case insensitive query with customer analyzer

### CODE REVIEW

**Comment by CCheSumo** [comment]
plox test

**Comment by rcoh** [comment]
travis is broken because of #20

**Comment by CCheSumo** [comment]
plox test

**Comment by CCheSumo** [comment]
Not sure how to trigger continuous-integration/travis-ci/push rebuild without owner permission. @rcoh would you be able to trigger a restart?

**Comment by rcoh** [comment]
@davidandrzej has the creds. Not sure if it's necessary.

File: elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClientTest.scala [hunk 1]
```diff
@@ -249,6 +259,22 @@ class RestlasticSearchClientTest extends WordSpec with Matchers with ScalaFuture
       whenReady(autocompelteUpper) {
         resp => resp should be(List("Case", "case"))
       }
+      // test special characters
+      val autocompelteSpecial = restClient.suggest(index, tpe, Suggest("#", Completion("suggest", 50, Map("f" -> "test"))))
```
Comment (rcoh) [inline elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClientTest.scala]: typo: autocomplete

Comment (CCheSumo) [inline elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClientTest.scala]: fixed

**Pull request merged**